content/ja/blog/_posts/2025-07-03-devices-failure-handling/index.md

```diff
@@ -0,0 +1,319 @@
+このオーケストレーターは、他のPodで障害が起こった為にジョブを再起動する必要がある場合に、メイン実行可能ファイルを再起動できるようになります。
+
+このようなソリューションは非常に脆弱で複雑です。
+大規模なトレーニングジョブで使用する場合、通常のJobSetの削除/再作成サイクルと比較して、コスト削減の価値がある場合が多いです。
+Kubernetesに新しいフックと拡張ポイントを開発することで、これらのソリューションの脆弱性を低減し効率化すれば、より小さなジョブにも簡単に適用でき、すべての人に利益をもたらします。
+
+### 故障モード: デバイスの劣化 {#failure-modes-device-degradation}
+
+すべてのデバイス障害が、ワークロード全体やバッチジョブにとって致命的であるとは限りません。
+ハードウェアスタックがますます複雑になるにつれて、ハードウェアスタックレイヤーのいずれかの設定ミスやドライバーの障害により、機能しているもののパフォーマンスが低下しているデバイスが生じる可能性があります。
+遅くなっている1つのデバイスが、トレーニングジョブ全体を遅くする可能性があります。
+
+このようなケースの報告はますます増えています。
+Kubernetesには今日、このタイプの障害を表現する方法がなくこれが最新のタイプの故障モードであるため、ハードウェアベンダーから検出に関するベストプラクティスやこれらの状況の修復に関するサードパーティツールがほとんど提供されていません。
+
+通常、これらの障害は、観察されたワークロード特性に基づいて検出されます。たとえば、特定のハードウェアでのAI/MLトレーニングステップの期待速度などです。
+これらの問題の修復は、ワークロードのニーズに大きく依存します。
+
+## ロードマップ {#roadmap}
+
+上のセクションで概説したように、Kubernetesは多くの拡張ポイントを提供しており、それらを使用して様々なDIYソリューションを実装できます。
+AI/ML の分野は、要件や使用パターンの変化に伴い、非常に急速に発展しています。
+SIG Nodeは、特定のシナリオをサポートするための新しいセマンティクスの導入よりも、ワークロード固有のシナリオを実装するための拡張ポイントを増やすという慎重なアプローチを採用しています。
+これは、一部のワークロードにしか適さない可能性のある障害に対する自動修復の実装よりも、障害に関する情報を容易に利用できるようにすることを優先するということです。
+
+このアプローチによりワークロード処理に劇的な変更が加えられることがなくなり、既存の円滑なDIYソリューションや従来のワークロードの体験が損なわれることはありません。
+
+今日使用されている多くのエラーハンドリング技術はAI/MLで機能しますが、非常に高コストです。
+SIG Nodeは、AI/MLのコスト削減が重要であるという理解の下で、これらをより安価にするための拡張ポイントに投資します。
+
+以下は、さまざまな故障モードに対して私たちが想定している具体的な投資事項です。
+
+### 故障モードのロードマップ: K8sインフラストラクチャ {#roadmap-for-failure-modes-k8s-infrastructure}
+
+Kubernetesインフラストラクチャの領域は理解しやすく、デバイスプラグインからDRAへの今後の移行を正しく行うために非常に重要です。
+SIG Nodeはこの領域で多くの作業項目を追跡しており、特に以下のものがあります:
+
+* [integrate kubelet with the systemd watchdog · Issue
+  #127460](https://github.com/kubernetes/kubernetes/issues/127460)
+* [DRA: detect stale DRA plugin sockets · Issue
+ #128696](https://github.com/kubernetes/kubernetes/issues/128696)
+* [Support takeover for devicemanager/device-plugin · Issue
+ #127803](https://github.com/kubernetes/kubernetes/issues/127803)
+* [Kubelet plugin registration reliability · Issue
+ #127457](https://github.com/kubernetes/kubernetes/issues/127457)
+* [Recreate the Device Manager gRPC server if failed · Issue
+ #128167](https://github.com/kubernetes/kubernetes/issues/128167)
+* [Pods that consume "devices" via Device Plugin always fail when Node reboots even if it implements plugins_registry interface · Issue
+  #128043](https://github.com/kubernetes/kubernetes/issues/128043)
+
+基本的に、Kubernetesコンポーネントのすべての相互作用は、kubeletの改善またはプラグイン開発とデプロイメントのベストプラクティスを介して信頼性が確保される必要があります。
+
+### 故障モードのロードマップ: デバイス故障 {#roadmap-for-failure-modes-device-failed}
+
+デバイス障害については、Kubernetesがサポートできる共通のシナリオでいくつかのパターンが既に現れています。
+しかし、最初のステップは、故障したデバイスに関する情報をより簡単に利用できるようにすることです。
+ここでの最初のステップは、[KEP 4680](https://kep.k8s.io/4680)(Add Resource Health Status to the Pod Status for Device Plugin and DRA)での作業です。
+
+テストされる長期的なアイデアには以下が含まれます:
+
+* デバイス障害をPod失敗に統合する。
+* ノードローカルのリトライポリシー、`restartPolicy=OnFailure`を持つPodに対してPod失敗ポリシーを有効にし、さらにそれ以上も対応可能にする。
+* `restartPolicy: Always`を含む、Podを _deschedule_ する機能。これにより、新しいデバイスを割り当てることができる。
```

Comment:
```suggestion
* Podを _スケジュール解除_ する機能。これは`restartPolicy: Always`を設定した場合を含むため、新しいデバイスを割り当てることができる。
```
> Ability to _deschedule_ pod, including with the `restartPolicy: Always`, so it can get a new device allocated.